### PR TITLE
task/279 credentials exchange types

### PR DESCRIPTION
**Pull request by JGiter**
### Summary

|             |   |
|-------------|---|
| Description |   |
| Jira issue  | https://energyweb.atlassian.net/browse/ICL-279 |

#### Type of request

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by nichonien — DISMISSED** [review]

File: packages/credentials-interface/src/exchange.types.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
+export type ExchangeInvitation = {
+  type: string;
+  body: {
+    /**
+     * exchange base url
+     */
+    url: string;
+  };
+};
```
Comment (jrhender) [inline packages/credentials-interface/src/exchange.types.ts:9]: Sorry @JGiter I know this is the format that I put in the story. However, this was the "to be" format based on on https://energyweb.atlassian.net/browse/REBDEV-89. The actual current format is:
```
"presentationLink": {
    "type": "string",
    "url": "string"
  }
```
Maybe to keep ICL more flexible and avoid needing to change ew-did-registry if the format changes, we can keep `ExchangeInvitation` and `EWF_EXCHANGE_INVITATION` in ICL for now?

Comment (JGiter) [reply packages/credentials-interface/src/exchange.types.ts]: no problem. from one side its make sense to put exchange types along with exchange logic (in iam-lib), but from the other these types are used in both iam-lib and ssi-hub. Besides it is potentially possible that there will be some relationships between exchange and credentials type

Comment (jrhender) [reply packages/credentials-interface/src/exchange.types.ts]: Ah good point, maybe best to have the type in ew-did-registry then 👍

File: packages/credentials-interface/src/exchange.types.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  limit_disclosure: 'required' | 'preferred';
+};
+
+export type Field = {
+  path: string[];
+  id?: string;
+  purpose?: string;
+  filter?: Filter;
+};
+
```
Comment (jrhender) [inline packages/credentials-interface/src/exchange.types.ts:108]: I would consider putting these in a separate file (maybe called `presentation-exchange`) in order to group these types as they all come from the Presentation Exchange spec https://identity.foundation/presentation-exchange/

**Review by jrhender — APPROVED** [review]
Looks good, just an organizational comment

File: packages/credentials-interface/src/exchange.types.ts (hunk 2 above)
Comment (JGiter) [reply packages/credentials-interface/src/exchange.types.ts:108]: Ok. lets keep file size under control

**Pull request merged**